### ISSUE TITLE
Configuration won't get deleted when its namespace is being deleted.

### ISSUE THREAD

**Issue by zzxwill** [kubevela/terraform-controller#329]
A Configuration won't get deleted when its namespace is being deleted. Please look into the details of this issue, please refer to https://cloud-native.slack.com/archives/C01BLQ3HTJA/p1656829163511879.

Here are the steps to reproduce the issue.

- Apply the Configuration 

```yaml
apiVersion: terraform.core.oam.dev/v1beta2
kind: Configuration
metadata:
  name: random-e2e
  namespace: n1
spec:
  hcl: |
    resource "random_id" "server" {
      byte_length = 8
    }
    
    output "random_id" {
      value = random_id.server.hex
    }

  inlineCredentials: true

  writeConnectionSecretToRef:
    name: random-conn
    namespace: default
```

- When the Configuration becomes ready, delete namespace `n1`.

The deletion of the Configuration got stuck and there are issues in the log of Terraform Controller.

```shell
apiVersion: terraform.core.oam.dev/v1beta2
kind: Configuration
metadata:
  name: random-e2e
  namespace: n1
spec:
  hcl: |
    resource "random_id" "server" {
      byte_length = 8
    }
    
    output "random_id" {
      value = random_id.server.hex
    }

  inlineCredentials: true

  writeConnectionSecretToRef:
    name: random-conn
    namespace: default
```

**Comment by chivalryq** [comment]
This will be fixed after #295, #332, #336

**Comment by rajramo61** [comment]
@chivalryq I noticed all 3 above issues are fixed and merged to master. Can this item be taken up next?

**Comment by chivalryq** [comment]
> @chivalryq I noticed all 3 above issues are fixed and merged to master. Can this item be taken up next?

Sorry for the delay, I have to do more test on feature about `--controller-namespace`, especially how the controller will behave when restarting with this arguments.

New PR #338 added e2e test for this scenario and fix some GC problems. It is become more reliable now. I need some help to test this feature with more case. So I'll make a release.

IMO `--controller-namespace` argument is still in alpha phase. More test should be added.
